platform/src/test/kotlin/org/stellar/anchor/platform/callback/RestRateIntegrationTest.kt

```diff
@@ -0,0 +1,273 @@
+      }
+    assertEquals(
+      "'sell_amount' is missing or not a positive number in the GET /rate response",
+      ex.message,
+    )
+
+    // Bad buy amount
+    rateResponse.rate.sellAmount = "100"
+    rateResponse.rate.buyAmount = badAmount
+    ex =
+      assertThrows<ServerErrorException> {
+        rateIntegration.validateRateResponse(request, rateResponse)
+      }
+    assertEquals(
+      "'buy_amount' is missing or not a positive number in the GET /rate response",
+      ex.message,
+    )
+  }
+
+  @Test
+  fun `test mis-matched sell_amount and buy_amount`() {
+    // Bad sell amount
+    rateResponse.rate.sellAmount = "100.01"
+    var ex =
+      assertThrows<ServerErrorException> {
+        rateIntegration.validateRateResponse(request, rateResponse)
+      }
+    assertEquals(
+      "'sell_amount' is not equal to price * buy_amount + (fee?:0) in the GET /rate response",
+      ex.message,
+    )
+
+    rateResponse.rate.sellAmount = "100.00"
+    rateResponse.rate.buyAmount = "94.00"
+
+    ex =
+      assertThrows<ServerErrorException> {
+        rateIntegration.validateRateResponse(request, rateResponse)
+      }
+    assertEquals(
+      "'sell_amount' is not equal to price * buy_amount + (fee?:0) in the GET /rate response",
+      ex.message,
+    )
+  }
+
+  @Test
+  fun `test bad fee total and asset`() {
+    rateResponse.rate.fee.total = null
+    var ex =
+      assertThrows<ServerErrorException> {
+        rateIntegration.validateRateResponse(request, rateResponse)
+      }
+    assertEquals(
+      "'fee.total' is missing or not a positive number in the GET /rate response",
+      ex.message,
+    )
+
+    rateResponse.rate.fee.total = "1.00"
+    rateResponse.rate.fee.asset = "unknown"
+    ex =
+      assertThrows<ServerErrorException> {
+        rateIntegration.validateRateResponse(request, rateResponse)
+      }
+    assertEquals(
+      "'fee.asset' is missing or not a valid asset in the GET /rate response",
+      ex.message,
+    )
+  }
+
+  @ParameterizedTest
+  @ValueSource(ints = [0, 1])
+  fun `test bad fee details without name or amount`(feeIndex: Int) {
+    rateResponse.rate.fee.details[feeIndex].name = null
+    var ex =
+      assertThrows<ServerErrorException> {
+        rateIntegration.validateRateResponse(request, rateResponse)
+      }
+    assertEquals(
+      "'fee.details.description[?].name' is missing in the GET /rate response",
+      ex.message,
+    )
+
+    rateResponse.rate.fee.details[feeIndex].name = "Sell fee"
+    rateResponse.rate.fee.details[feeIndex].amount = null
+    ex =
+      assertThrows<ServerErrorException> {
+        rateIntegration.validateRateResponse(request, rateResponse)
+      }
+    assertEquals(
+      "'fee.details[?].description.amount' is missing or not a positive number in the GET /rate response",
+      ex.message,
+    )
+
+    rateResponse.rate.fee.details[feeIndex].name = "Sell fee"
+    rateResponse.rate.fee.details[feeIndex].amount = "0.71"
```

Comment:
Can we add a comment with what the expected amount should be, and why?

Reply:
Added a comment.